plugins/shadow-block-converter/src/shadow_block_converter.ts

```diff
@@ -228,24 +239,32 @@ function reifyEditedShadowBlock(shadowBlock: Block): Blockly.Block | null {
 
   // If the parent block is a shadow, it must be converted first.
   if (parentBlock.isShadow()) {
-    const newParentBlock: Blockly.Block | null =
-      reifyEditedShadowBlock(parentBlock);
+    const newParentBlock = reifyEditedShadowBlock(parentBlock);
     if (newParentBlock === null) {
-      // No parent block was created, so we can't recreate the current block
-      // either.
-      return null;
+      throw new Error(
+        "No parent block was created, so we can't recreate the " +
+          'current block either.',
+      );
     }
     parentBlock = newParentBlock;
 
     // The reference to the connection is obsolete. Find it from the new parent.
-    const parentConnections: Blockly.Connection[] =
-      parentBlock.getConnections_(true);
-    if (connectionIndex == -1 || connectionIndex >= parentConnections.length) {
-      // Couldn't find the corresponding connection on the new version of the
-      // parent block.
-      return null;
+    if (inputIndexInParent === null) {
+      parentConnection = parentBlock.nextConnection;
+    } else if (
+      inputIndexInParent == -1 ||
```

Comment:
```suggestion
      inputIndexInParent < 0 ||
```
(Or if that is not semantically correct, then at least change to `… === -1`.)

Reply:
Done.